### PR TITLE
Feature/cc 103 template audit

### PR DESCRIPTION
**Pull request by ZhihanChenZack**
Add Template management page
The template management page includes two subpages:
-Template List: list all templates in the database.
-Template audit: list all censoring templates in the database and allows the admin to Publish or Reject censoring templates

image

Other functions:
-Template review image popup modal.
-Admin publish and reject operation confirmation popup modal.
image

image

### CODE REVIEW

File: src/redux/api/templateApi.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { ITemplate, ITemplateDataDb } from "@/interfaces/template";
+import _ from "lodash";
+
+export const templateApi = createApi({
+  reducerPath: "templates",
+  baseQuery: fetchBaseQuery({ baseUrl: environment.apiBaseUrl }),
+  tagTypes: ["template"],
+  endpoints: (builder) => ({
+    getTemplates: builder.query<Omit<ITemplateDataDb, "__v" | "isDeleted">[], {userId? : string | void, status : string | void}>({
+      query: ({userId, status}) => (userId ? `/templates?user_id=${userId}` : `/templates?status=${status}`),
```
[the remark this answers is not in the document]
Comment (ZhihanChenZack) [reply src/redux/api/templateApi.tsx:12]: reserved for userID search

**Review by Double-Twelve — APPROVED** [review]
Zack, how incredible you are and your work! Thank you for a job well done!

**Review by Cecilia0109 — APPROVED** [review]
LGTM

**Review by STY1997 — APPROVED** [review]
lgtm

**Pull request merged**